### PR TITLE
Repositories improvements (DEV)

### PR DESCRIPTION
**Pull request by jurajkusnier**
Sync `RecoveryCertificateRepository`, `TestCertificateRepository` and `VaccinationCertificateRepository` to use  `Map<containerId,container>`  structure to represent `internalData`.
With Map, we can be sure that there will be always only 1 certificate for any QR code.
This is not the case for Set, where the collection cannot contain duplicate elements, but as the container consists not only of QR code but also other data, we can't be sure there are no containers with the same QR code.

### CODE REVIEW

**Review by chiljamgossow — APPROVED** [review]
Looks much better now, thanks!

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/repository/storage/VaccinationStorage.kt [hunk 1]
```diff
@@ -37,8 +37,6 @@ class VaccinationStorage @Inject constructor(
                 prefs.getString(PKEY_VACCINATION_CERT, null) ?: return emptySet(),
                 TYPE_TOKEN
             )
-            .associateBy { it.vaccinationQrCode }
-            .map { it.value }
             .toSet()
```
Comment (LukasLechnerDev) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/repository/storage/VaccinationStorage.kt:40]: `.toSet()` could be removed here, or?

**Review by LukasLechnerDev — APPROVED** [review]
👍 only one comment

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/repository/storage/VaccinationStorage.kt (hunk 1 above)
Comment (jurajkusnier) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/repository/storage/VaccinationStorage.kt:40]: could be

**Pull request merged**